### PR TITLE
Update license

### PR DESCRIPTION
**Pull request by hanlint**
Based on feedback in https://github.com/mosaicml/composer/pull/1007, this PR:
* Updates all license headers with the Apache 2.0 SPDX, and changes the copyright to include all composer authors.
```
# Copyright 2022 MosaicML Composer authors
# SPDX-License-Identifier: Apache-2.0
```
* moves the `COPYRIGHT_HEADER` to `.ci/` to allow github license auto-detection to work.

### CODE REVIEW

File: .ci/FILE_HEADER [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Copyright 2022 MosaicML Composer authors
+SPDX-License-Identifier: Apache-2.0
```
Comment (ravi-mosaicml) [inline .ci/FILE_HEADER:2]: nit: newline
```suggestion
SPDX-License-Identifier: Apache-2.0

```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -13,58 +13,58 @@ repos:
...
-#     hooks:
-#         - id: pylint
-#           entry: pylint
-#           args: ['composer', 'examples', 'tests']
-#           language: python
-#           types: [python]
-#           require_serial: true
-#     rev: v2.12.2
--   repo: local
+  # -   repo: https://github.com/pycqa/pylint
```
Comment (ravi-mosaicml) [inline .pre-commit-config.yaml:20]: Curious about these formatting changes? Is this pre-commit running on itself? :rofl:

**Review by ravi-mosaicml — APPROVED** [review]
LGTM

File: .ci/FILE_HEADER (hunk 1 above)
Comment (hanlint) [reply .ci/FILE_HEADER:2]: I dont think this is needed -- the previous header did not have a new line, and testing on a blank file, looks like the pre-commit works without.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (hanlint) [reply .pre-commit-config.yaml:20]: Might be own VSCode yaml formatting

**Comment by mbrukman** [comment]
@hanlint — thank you for making these changes! I see the repo is now correctly marked as "Apache 2.0" on the main page, so great job! 🎉

**Pull request merged**